modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -566,6 +573,8 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
                             resp.code = "200";
                         }
 
+                        resp.baseType = getNonMutableContainerTypeIfNeeded(resp.baseType);
```

Comment:
@rm3l could you please explain what this does and why it's needed? Thanks

Reply:
Similar reasons as in my [other comment below](https://github.com/OpenAPITools/openapi-generator/pull/11154#discussion_r793094727).

For example, given the following input operation:
```yaml
/store/inventory:
    get:
      tags:
        - store
      summary: Returns pet inventories by status
      description: Returns a map of status codes to quantities
      operationId: getInventory
      produces:
        - application/json
      parameters: []
      responses:
        '200':
          description: successful operation
          schema:
            type: object
            additionalProperties:
              type: integer
              format: int32
```

Without the changes here, this resulted in the following code generated by the `kotlin-spring` generator:

- StoreApiController.kt : the Response field in the `ApiResponse` annotation has changed to `MutableMap`, which, I guess, is not valid.

```diff
diff --git a/samples/server/petstore/kotlin-springboot-modelMutable/src/main/kotlin/org/openapitools/api/StoreApiController.kt b/samples/server/petstore/kotlin-springboot-modelMutable/src/main/kotlin/org/openapitools/api/StoreApiController.kt
index e2be73b4ec3..ad63efeae2d 100644
--- a/samples/server/petstore/kotlin-springboot-modelMutable/src/main/kotlin/org/openapitools/api/StoreApiController.kt
+++ b/samples/server/petstore/kotlin-springboot-modelMutable/src/main/kotlin/org/openapitools/api/StoreApiController.kt
@@ -58,7 +58,7 @@ class StoreApiController(@Autowired(required = true) val service: StoreApiServic
         responseContainer = "Map",
         authorizations = [Authorization(value = "api_key")])
     @ApiResponses(
        value = [ApiResponse(code = 200, message = "successful operation", 
-       response = kotlin.collections.Map::class,
        responseContainer = "Map")])
        value = [ApiResponse(code = 200, message = "successful operation",
+       response = kotlin.collections.MutableMap::class,
        responseContainer = "Map")])
     @RequestMapping(
         method = [RequestMethod.GET],
         value = ["/store/inventory"],
```

Again, maybe this is not the right approach here or maybe I missed something, but I found no clear way to distinguish between the type mappings used for generating models (Kotlin data classes) and response types in the `kotlin-spring` generator. Also, since the original issue I reported was more related to model mutability (and not, I thought it would be ok not to change the code generated for the Spring Controllers.

Reply:
Thanks for explaining, this way we have an explanation in case we need it 👍